lib/action.ts

```diff
@@ -1,11 +1,50 @@
 import { Err, Ok } from "./result.ts";
 import type { Effect, Operation } from "./types.ts";
 
-interface Resolver<T> {
+/**
+ * A function used to define how an {@link action} will be resolved. This
+ * function is passed two arguments: a `resolve` callback used to resolve
+ * the action with a value and a `reject` callback used to reject the
+ * action with a provided error.
+ *
+ * @param executor - the function to run whenever an action is evaluated.
+ * @return a "finally" function that is _always_ run whether the
+ *           action is resolved, rejected, or discarded.
+ */
+interface Executor<T> {
```

Comment:
This was renamed to `Executor` because  that's the terminology used by the `Promise` constructor